### ISSUE TITLE
USWDS - Bug: Axe a11y error in pagination

### ISSUE THREAD

**Issue by mejiaj** [uswds/uswds#5022]
### Describe the bug

We use axe-core for accessibility testing on components. Versions above `5.4.0` introduce new rules (some based on WCAG 2.2). One new rule in particular is now reporting an error on the pagination component.

Originally reported in this [PR comment [USWDS - Deps: Update dependencies [minor and patch]]](https://github.com/uswds/uswds/pull/5019#issuecomment-1307796280)

### Steps to reproduce the bug

1. Update `axe-core` dependency to `>=5.4.0`
2. Run `npm run test:a11y`
3. Confirm a11y error

### Expected Behavior

There should be no errors.

### Related code

https://github.com/uswds/uswds/blob/69cf6929585b5c95ed9700a4502830f84e497e95/packages/usa-pagination/src/usa-pagination.twig#L5

### System setup

USWDS 3.2.0

### Additional context

We need to either remove presentation role or add `listitem` role to sibling links.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by mejiaj** [comment]
Adding @jejenka for a11y support and testing.

**Comment by mejiaj** [comment]
Reported again on on 03/10/23 in slack. Removing confirmation tag.
https://gsa-tts.slack.com/archives/C3F14AHSQ/p1678480430853149

**Comment by mgifford** [comment]
There is a related issue on Drupal.org [https://www.drupal.org/project/drupal/issues/3347326](https://www.drupal.org/project/drupal/issues/3347326)  
  
I think we can align with what the UK is doing here:   
https://design-system.service.gov.uk/components/pagination/

**Comment by thisisdano** [comment]
Seems like the simplest path forward is to remove the `presentation` role
